src/corehost/cli/deps_resolver.cpp

```diff
@@ -470,7 +470,8 @@ bool deps_resolver_t::resolve_probe_dirs(
         {
             // For standalone apps, apphost.exe will be renamed. Do not use the full package name
             // because of rid-fallback could happen (ex: CentOS falling back to RHEL)
-            if (ends_with(entry.library_name, _X(".Microsoft.NETCore.App"), false) && (entry.asset_name == _X("dotnet") || entry.asset_name == _X("apphost")))
+            if (((entry.asset_name == _X("dotnet")) && (ends_with(entry.library_name, _X(".Microsoft.NETCore.App"), false) || ends_with(entry.library_name, _X(".Microsoft.NETCore.DotNetHost"), false))) ||
```

Comment:
When does the `dotnet` asset come in the `Microsoft.NETCore.App` library anymore?  It should only come in the `Microsoft.NETCore.DotNetHost` library, right?